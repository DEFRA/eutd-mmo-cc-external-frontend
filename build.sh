#/bin/bash
curl -O https://nodejs.org/dist/v12.18.4/node-v12.18.4-linux-x64.tar.xz
tar -xf node-v12.18.4-linux-x64.tar.xz --directory /tmp
export PATH="/tmp/node-v12.18.4-linux-x64/bin:$PATH"
rm node-v12.18.4-linux-x64.tar.xz
npm ci
npm run test-vstack
rm -rf /tmp/node-v12.18.4-linux-x64