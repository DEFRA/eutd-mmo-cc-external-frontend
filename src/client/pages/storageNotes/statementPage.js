import React, { Fragment } from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import {
  Main,
  Header,
  Panel,
  GridRow,
  GridCol
} from 'govuk-react';

import {
  clearStorageNotes, clearTransportDetails,
  saveStorageNotes, saveStorageNotesToRedis
} from '../../actions';
import {
  getCompletedDocument
} from '../../actions/document.actions';
import PageTitle from '../../components/PageTitle';
import PageTemplateWrapper from '../../components/PageTemplateWrapper';
import HelpLink from '../../components/HelpLink';
import { monitorEvent } from '../../actions/monitor.actions';
import {withTranslation} from 'react-i18next';

class StatementPage extends React.Component {

  hasRequiredData() {
    const { completedDocument } = this.props;

    return completedDocument && completedDocument.documentNumber && completedDocument.documentUri && completedDocument.documentNumber.includes('-SD-');
  }

  async componentDidMount() {
    const documentNumber = this.props.match.params['documentNumber'];

    this.props.clear({ catches: [{}], storageFacilities: [{}], pdf: this.props.storageNotes.pdf });
    this.props.clearTransportDetails();

    await this.props.getCompletedDocument(documentNumber);

    if (!this.hasRequiredData()) {
      this.props.history.push('/create-storage-document/storage-documents');
    }
  }

  render() {
    const {route, completedDocument, t} = this.props;
    const documentNumber = (completedDocument && completedDocument.documentNumber) ? completedDocument.documentNumber : 'not found';
    const uri = (completedDocument && completedDocument.documentUri) ? completedDocument.documentUri :  'not found';

    const buildUrl = (docUrl) => {
      if(docUrl.indexOf('/') !== -1) {
        const splitUrlLength = docUrl.split('/').length;
        const splitUrl = docUrl.split('/')[splitUrlLength - 1];
        const docId = splitUrl.split('?')[0];
        return `/pdf/export-certificates/${docId}`;
      }
      return `/pdf/export-certificates/${docUrl}`;
    };

    const monitorViewEvent = doc => {
      const journey = this.props.route.journey;
      this.props.monitorEvent(doc, journey);
    };

    return (
      <Main>
        <PageTitle title={`${t('sdStatementPageTitle')} - ${t('sdCommonTitle')}` }/>
        <GridRow>
          <GridCol>
            <Panel
              className="submitted-certificate-bg"
              panelTitle={t('sdCreatedPanelTitle')}
              panelBody={
                <Fragment>
                  {t('sdCreatedPanelBody')}
                  <br />
                  <strong id="documentNumber">{documentNumber}</strong>
                </Fragment>
              }
            />
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <Header level={2} size="MEDIUM">{t('sdCreatedNextStepsHeader')}</Header>
            <ol className="list list-number">
              <li>
                <strong><a rel="noopener noreferrer" onClick={monitorViewEvent.bind(this, documentNumber)} href={buildUrl(uri)}>{t('sdCreatedDownloadLink')}</a>. </strong>
                {t('sdCreatedDownloadDocumentNotesSubHeading')}
                <ul className="list list-bullet">
                  <li>{t('sdCreatedDownloadDocumentNotesFirefox')}</li>
                  <li>{t('sdCreatedDownloadDocumentNotesMobile')}</li>
                </ul>
              </li>
              <li>
                <strong>{t('sdCreatedEmailToImporter')}. </strong>
                <span>{t('sdCreatedEmailToImporterText')}</span>
              </li>
            </ol>
            <p>
              <Link to={'/create-storage-document/storage-documents'}>{t('sdCreatedViewOrCreateNewLink')}</Link>
            </p>
            <HelpLink journey={route.journey}/>
          </GridCol>
        </GridRow>
      </Main>
    );
  }
}

function mapStateToProps(state) {
  return {
    storageNotes: state.storageNotes,
    documents: state.documents,
    completedDocument: state.completedDocument,
    monitorEvent: monitorEvent
  };
}

function loadData(store) {
  if (Object.keys(this.queryParams).length !== 0) {
    store.dispatch(saveStorageNotes({ pdf: { uri: this.queryParams.uri, documentNumber: this.queryParams.documentNumber } }));
  }
}

export const component = withRouter(connect(mapStateToProps,
  {
    save: saveStorageNotes,
    saveToRedis: saveStorageNotesToRedis,
    clear: clearStorageNotes,
    monitorEvent,
    clearTransportDetails,
    getCompletedDocument
    }
  )(withTranslation()(StatementPage))
);

export default {
  loadData,
  component: PageTemplateWrapper(component)
};
