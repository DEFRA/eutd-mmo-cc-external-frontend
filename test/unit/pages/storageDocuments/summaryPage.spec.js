import { mount } from 'enzyme/build';
import { Provider } from 'react-redux';
import { Router, Route, MemoryRouter } from 'react-router-dom';
import * as React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import {
  getStorageNotesFromRedis,
  getTransportDetails,
  getExporterFromMongo,
  saveStorageNotes,
  generateStorageNotesPdf
} from '../../../../src/client/actions';
import { render } from '@testing-library/react';

jest.mock('../../../../src/client/actions');

import { component as Summary } from '../../../../src/client/pages/storageNotes/summary';
const mockStore = configureStore([thunk]);

const store = mockStore({
  match: {
    params: {
      documentNumber: '123'
    }
  },
  global: {
    allFish: []
  },
  exporter: {
    model: {
      exporterCompanyName: 'MailOrderFish Ltd',
      addressOne: '5 Costa Road',
      townCity: 'Hebburn',
      postcode: 'NE1 4FS',
      _updated: true
    }
  },
  storageNotes: {
    addAnotherProduct: 'notset',
    addAnotherStorageFacility: 'notset',
    catches: [{
      product: 'Asda squid burgers',
      commodityCode: '123456',
      certificateNumber: '12345',
      productWeight: '1000',
      dateOfUnloading: '25/01/2019',
      placeOfUnloading: 'Dover',
      transportUnloadedFrom: 'Car',
      weightOnCC: '500'
    }],
    validationErrors: [{}],
    storageFacilities: [{
      facilityName: 'Test1 from store',
      facilityAddressOne: 'Some Address One',
      facilityTownCity: 'Some Town',
      facilityPostcode: 'Some PostCode',
      _facilityUpdated: false
    },
      {
        facilityName: 'Test2 from store',
        facilityAddressOne: 'Some Address One',
        facilityTownCity: 'Some Town',
        facilityPostcode: 'Some PostCode',
        _facilityUpdated: true
    }],
  },
  transport: {
    vehicle: 'truck',
    cmr: 'true',
    exportedTo: {
      isoCodeAlpha2: 'Alpha2',
      isoCodeAlpha3: 'Alpha3',
      isoNumericCode: 'IsoNumericCode',
      officialCountryName: 'Brazil',
    }
  },
  fish: [{}],
  saveAsDraft: {
    currentUri: {
      'GBR-2020-SD-7B3E62FE9': '/create-storage-document/:documentNumber/add-exporter-details'
    },
    journey: 'storageNotes',
    user_id: 'ABCD-EFGH-IJKL-MNOP-QRST-UVWX-YZ12'
  }
}
);


const props = {
  match: {
    params: {
      documentNumber: '123'
    }
  },
  path: '/create-storage-document/:documentNumber/check-your-information',
  url: '/create-storage-document/GBR-2021-SD-0467026EB/check-your-information',
  route: {
    journey: 'storageNotes',
    path: '/create-storage-document/:documentNumber/check-your-information',
    title: 'Create a UK storage document - GOV.UK',
  },
  errors: [],
  history: [],
  journey: 'processingStatement',
  addresses: [{
    building_name: 'LANCASTER HOUSE',
    city: 'NEWCASTLE UPON TYNE',
    country: 'ENGLAND',
    county: 'TYNESIDE',
    postCode: 'NE4 7YH',
    street_name: 'HAMPSHIRE COURT',

  }],
  global: {
    allFish: []
  },
  exporter: {
    model: {
      exporterFullName: 'Mr Fish Exporter',
      exportCompanyName: 'MailOrderFish Ltd',
      addressOne: '5 Costa Road',
      townCity: 'Hebburn',
      postcode: 'NE1 4FS',
      _updated: true
    }
  },
  storageNotes: {
    addAnotherProduct: 'notset',
    addAnotherStorageFacility: 'notset',
    catches: [{
      product: 'Asda squid burgers',
      commodityCode: '123456',
      certificateNumber: '12345',
      productWeight: '1000',
      dateOfUnloading: '25/01/2019',
      placeOfUnloading: 'Dover',
      transportUnloadedFrom: 'Car',
      weightOnCC: '500'
    }],
    global: {
      allFish: []
    },
    exporter: {
      model: {
        exporterCompanyName: 'MailOrderFish Ltd',
        addressOne: '5 Costa Road',
        townCity: 'Hebburn',
        postcode: 'NE1 4FS',
        _updated: true
      }
    },
    storageNotes: {
      addAnotherProduct: 'notset',
      addAnotherStorageFacility: 'notset',
      catches: [{
        product: 'Asda squid burgers',
        commodityCode: '123456',
        certificateNumber: '12345',
        productWeight: '1000',
        dateOfUnloading: '25/01/2019',
        placeOfUnloading: 'Dover',
        transportUnloadedFrom: 'Car',
        weightOnCC: '500'
      }],
      validationErrors: [{}],
      storageFacilities: [{
        facilityName: 'Test1',
        facilityAddressOne: 'Some Address One',
        facilityTownCity: 'Some Town',
        facilityPostcode: 'Some PostCode',
        _facilityUpdated: false
      },
      {
        facilityName: 'Test2',
        facilityAddressOne: 'Some Address One',
        facilityTownCity: 'Some Town',
        facilityPostcode: 'Some PostCode',
        _facilityUpdated: true
      }],
    },
    transport: {
      vehicle: 'truck',
      cmr: 'true',
      nationalityOfVehicle: 'some-of-vehicle',
      registrationNumber: 'registration-number',
      departurePlace: 'departure-palce',
      exportDate: 'export-date',
      exportedTo: {
        isoCodeAlpha2: 'Alpha2',
        isoCodeAlpha3: 'Alpha3',
        isoNumericCode: 'IsoNumericCode',
        officialCountryName: 'Brazil',
      }
    },
  },
  transport: {
    vehicle: 'truck',
    cmr: 'true',
    exportedTo: {
      isoCodeAlpha2: 'Alpha2',
      isoCodeAlpha3: 'Alpha3',
      isoNumericCode: 'IsoNumericCode',
      officialCountryName: 'Brazil',
    }
  },
  fish: [{}],
  saveAsDraft: {
    currentUri: {
      'GBR-2020-SD-7B3E62FE9': '/create-storage-document/:documentNumber/add-exporter-details'
    },
    journey: 'storageNotes',
    user_id: 'ABCD-EFGH-IJKL-MNOP-QRST-UVWX-YZ12'
  },

};

describe('SummaryPage', () => {
  const documentNumber = 'GBR-2020-SD-123456';
  const history = createMemoryHistory({ initialEntries: [`/create-storage-document/${documentNumber}/check-your-information`] });
  const mockHistoryPush = jest.spyOn(history, 'push');

  let wrapper;
  beforeAll(() => {
    getExporterFromMongo.mockReturnValue({ type: 'GET_STORAGE_NOTES_FROM_REDIS' });
    getStorageNotesFromRedis.mockReturnValue({ type: 'GET_STORAGE_NOTES_FROM_REDIS' });
    getTransportDetails.mockReturnValue({ type: 'GET_TRANSPORT_DETAILS ' });
    saveStorageNotes.mockReturnValue({ type: 'SAVE_STORAGE_NOTES' });
    generateStorageNotesPdf.mockReturnValue({ type: 'GENERAGE_STORAGE_NOTES' });

    wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
            <Route path='/create-storage-document/:documentNumber/check-your-information' exact match={{ params: { documentNumber: '123' } }} >
              <Summary {...props} />
            </Route>
        </Router>
      </Provider>
    );
  });

  afterAll(() => {
    getExporterFromMongo.mockReset();
    getStorageNotesFromRedis.mockReset();
    getTransportDetails.mockReset();
    saveStorageNotes.mockReset();
    generateStorageNotesPdf.mockReset();
  });

  it('should redirect to add storage facilities page if _facilityUpdated is true', () => {
    expect(wrapper).toBeDefined();
    expect(mockHistoryPush).toHaveBeenCalledWith('/create-storage-document/GBR-2020-SD-123456/add-storage-facility-details/1');
  });
});

describe('Summary page for sd journey',() => {

  const documentNumber = 'GBR-2020-SD-123456';
  const history = createMemoryHistory({ initialEntries: [`/create-storage-document/${documentNumber}/check-your-information`] });


  let wrapper;
  beforeEach(() => {
    getExporterFromMongo.mockReturnValue({ type: 'GET_STORAGE_NOTES_FROM_REDIS' });
    getStorageNotesFromRedis.mockReturnValue({ type: 'GET_STORAGE_NOTES_FROM_REDIS' });
    getTransportDetails.mockReturnValue({ type: 'GET_TRANSPORT_DETAILS ' });
    saveStorageNotes.mockReturnValue({ type: 'SAVE_STORAGE_NOTES' });
  });
  afterEach(() => {
    getExporterFromMongo.mockReset();
    getStorageNotesFromRedis.mockReset();
    getTransportDetails.mockReset();
    saveStorageNotes.mockReset();
  });


  wrapper = mount(
    <Provider store={store}>
      <Router history={history}>
          <Route path='/create-storage-document/:documentNumber/check-your-information' exact match={{ params: { documentNumber: '123' } }} >
            <Summary {...props} />
          </Route>
      </Router>
    </Provider>
  );


  it('should take a snapshot of the whole page', () => {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });
});

describe('SummaryPage - error island', () => {

  const props = {
    match: {
      params: {
        documentNumber: '123'
      }
    },
    path: '/create-storage-document/:documentNumber/check-your-information',
    url: '/create-storage-document/GBR-2021-SD-0467026EB/check-your-information',
    route: {
      journey: 'storageNotes',
      path: '/create-storage-document/:documentNumber/check-your-information',
      title: 'Create a UK storage document - GOV.UK',
    },
    errors: [],
    history: [],
    journey: 'processingStatement',
    addresses: [{
      building_name: 'LANCASTER HOUSE',
      city: 'NEWCASTLE UPON TYNE',
      country: 'ENGLAND',
      county: 'TYNESIDE',
      postCode: 'NE4 7YH',
      street_name: 'HAMPSHIRE COURT',

    }],
    global: {
      allFish: []
    },
    exporter: {
      model: {
        exporterCompanyName: 'MailOrderFish Ltd',
        addressOne: '5 Costa Road',
        townCity: 'Hebburn',
        postcode: 'NE1 4FS',
        _updated: true
      }
    },
    storageNotes: {
      addAnotherProduct: 'notset',
      addAnotherStorageFacility: 'notset',
      catches: [{
        product: 'Asda squid burgers',
        commodityCode: '123456',
        certificateNumber: '12345',
        productWeight: '1000',
        dateOfUnloading: '25/01/2019',
        placeOfUnloading: 'Dover',
        transportUnloadedFrom: 'Car',
        weightOnCC: '500'
      }],
      validationErrors: [{}],
      storageFacilities: [{
        facilityName: 'Test1',
        facilityAddressOne: 'Some Address One',
        facilityTownCity: 'Some Town',
        facilityPostcode: 'Some PostCode',
        _facilityUpdated: false
      },
      {
        facilityName: 'Test2',
        facilityAddressOne: 'Some Address One',
        facilityTownCity: 'Some Town',
        facilityPostcode: 'Some PostCode',
        _facilityUpdated: true
      }],
    },
    transport: {
      vehicle: 'truck',
      cmr: 'true',
      nationalityOfVehicle: 'some-of-vehicle',
      registrationNumber: 'registration-number',
      departurePlace: 'departure-palce',
      exportDate: 'export-date',
      exportedTo: {
        isoCodeAlpha2: 'Alpha2',
        isoCodeAlpha3: 'Alpha3',
        isoNumericCode: 'IsoNumericCode',
        officialCountryName: 'Brazil',
      }
    },
    fish: [{}],
    saveAsDraft: {
      currentUri: {
        'GBR-2020-SD-7B3E62FE9': '/create-storage-document/:documentNumber/add-exporter-details'
      },
      journey: 'storageNotes',
      user_id: 'ABCD-EFGH-IJKL-MNOP-QRST-UVWX-YZ12'
    },

  };

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <Summary {...props}/>
      </MemoryRouter>
    </Provider>
  );

  it('should not render an error island component', () => {
    expect(wrapper.find('ErrorIsland').exists()).toBe(false);
  });

  it('should not render an error island component when validation errors are empty', () => {

    wrapper.find('form').simulate(
      'submit',
      {preventDefault() {}}
    );

    expect(wrapper.find('ErrorIsland').exists()).toBe(false);
  });
});