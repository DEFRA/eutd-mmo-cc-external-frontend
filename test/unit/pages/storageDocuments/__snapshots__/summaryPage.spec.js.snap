// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary page for sd journey should take a snapshot of the whole page 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/storageNotes/back?n=/create-storage-document/GBR-2020-SD-123456/do-you-have-a-road-transport-document"
          >
            Back
          </a>
          <div
            class="notification-banner"
            role="alert"
            style="border: 0px;"
          >
            <div
              class="notification-banner__header"
            >
              <h2
                class="notification-banner__title"
              >
                Important
              </h2>
            </div>
            <div
              class="notification-banner__content"
            >
              <p
                class="notification-banner__heading"
              >
                Due to improvements in the way addresses are managed, the exporterâ€™s address in this document has been reloaded from your Defra account. Please check the address is correct and change if necessary
              </p>
            </div>
          </div>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            Check your information before you create the storage document
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/generatePdf?n=/create-storage-document/GBR-2020-SD-123456/storage-document-created"
        method="POST"
      >
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Document details
            </h2>
            <dl
              class="css-39qxwj"
            >
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Document number
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  GBR-2020-SD-123456
                </dd>
              </div>
            </dl>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Exporter details
            </h2>
            <dl
              class="summary css-39qxwj"
              id="exporter-details"
            >
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Company name
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  MailOrderFish Ltd
                </dd>
                <span
                  class="css-3ay1af"
                >
                  <a
                    class="css-80kuzi-StyledHoc e109g54z0"
                    href="/create-storage-document/GBR-2020-SD-123456/add-exporter-details"
                    id="change-exporterCompanyName"
                  >
                    Change
                    <span
                      class="govuk-visually-hidden"
                    >
                      exporter details
                    </span>
                  </a>
                </span>
              </div>
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Company address
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  5 Costa Road
                  <br />
                  Hebburn
                  <br />
                  NE1 4FS
                  <br />
                </dd>
                <span
                  class="css-3ay1af"
                />
              </div>
            </dl>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Products in this consignment
            </h2>
            <div
              class="catch-row "
              id="validationError"
            >
              <div
                class="catch-table"
              >
                <dl
                  class="css-39qxwj"
                >
                  <div
                    class="css-fvyygl"
                  >
                    <dt
                      class="css-umwrei"
                    >
                      Asda squid burgers
                    </dt>
                    <dd
                      class="css-1szlssk"
                    />
                    <span
                      class="css-jgj7td"
                    >
                      <a
                        class="css-80kuzi-StyledHoc e109g54z0"
                        href="/create-storage-document/GBR-2020-SD-123456/add-product-to-this-consignment/0"
                        id="change-catches-0-product"
                      >
                        Change
                        <span
                          class="govuk-visually-hidden"
                        >
                          product Asda squid burgers
                        </span>
                      </a>
                    </span>
                  </div>
                  <div
                    class="css-fvyygl"
                  >
                    <dd
                      class="css-1szlssk"
                    >
                      Commodity code
                    </dd>
                    <dd
                      class="css-1szlssk"
                    >
                      123456
                    </dd>
                    <span
                      class="css-jgj7td"
                    />
                  </div>
                  <div
                    class="css-fvyygl"
                  >
                    <dd
                      class="css-1szlssk"
                    >
                      Export weight
                    </dd>
                    <dd
                      class="css-1szlssk"
                    >
                      1000
                      kg
                    </dd>
                    <span
                      class="css-jgj7td"
                    />
                  </div>
                  <div
                    class="css-fvyygl"
                  >
                    <dd
                      class="css-1szlssk"
                    >
                      UK entry:
                       
                      Date
                    </dd>
                    <dd
                      class="css-1szlssk"
                    >
                      25 January 2019
                    </dd>
                    <span
                      class="css-jgj7td"
                    />
                  </div>
                  <div
                    class="css-16k4qo5"
                  >
                    <dd
                      class="css-1szlssk"
                    >
                      UK entry:
                       
                      Place
                    </dd>
                    <dd
                      class="css-1szlssk"
                    >
                      Dover
                    </dd>
                    <span
                      class="css-jgj7td"
                    />
                  </div>
                  <div
                    class="css-16k4qo5"
                  >
                    <dd
                      class="css-1szlssk"
                    >
                      UK entry:
                       
                      Transport details
                    </dd>
                    <dd
                      class="css-1szlssk"
                    >
                      Car
                    </dd>
                    <span
                      class="css-jgj7td"
                    />
                  </div>
                  <div
                    class="css-16k4qo5"
                  >
                    <dd
                      class="css-1szlssk"
                    >
                      UK entry:
                       
                      Document number
                    </dd>
                    <dd
                      class="css-1szlssk"
                    >
                      12345
                    </dd>
                    <span
                      class="css-jgj7td"
                    />
                  </div>
                  <div
                    class="css-16k4qo5"
                  >
                    <dd
                      class="css-1szlssk"
                    >
                      UK entry:
                       
                      Weight on document
                    </dd>
                    <dd
                      class="css-1szlssk"
                    >
                      500
                      kg
                    </dd>
                    <span
                      class="css-jgj7td"
                    />
                  </div>
                </dl>
              </div>
            </div>
            <div
              class="seperationLine"
            />
            <br />
          </div>
        </div>
        <h2
          class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
        >
          Storage facility details
        </h2>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <dl
              class="summary css-39qxwj"
            >
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Facility name
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  Test1 from store
                </dd>
                <span
                  class="css-3ay1af"
                >
                  <a
                    class="css-80kuzi-StyledHoc e109g54z0"
                    href="/create-storage-document/GBR-2020-SD-123456/add-storage-facility-details/0"
                    id="change-storageFacilities-0-facilityName"
                  >
                    Change
                    <span
                      class="govuk-visually-hidden"
                    >
                      storage facility Test1 from store
                    </span>
                  </a>
                </span>
              </div>
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Address
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  Some Address One
                  <br />
                  Some Town
                  <br />
                  Some PostCode
                  <br />
                </dd>
                <span
                  class="css-3ay1af"
                />
              </div>
            </dl>
          </div>
        </div>
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <dl
              class="summary css-39qxwj"
            >
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Facility name
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  Test2 from store
                </dd>
                <span
                  class="css-3ay1af"
                >
                  <a
                    class="css-80kuzi-StyledHoc e109g54z0"
                    href="/create-storage-document/GBR-2020-SD-123456/add-storage-facility-details/1"
                    id="change-storageFacilities-1-facilityName"
                  >
                    Change
                    <span
                      class="govuk-visually-hidden"
                    >
                      storage facility Test2 from store
                    </span>
                  </a>
                </span>
              </div>
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Address
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  Some Address One
                  <br />
                  Some Town
                  <br />
                  Some PostCode
                  <br />
                </dd>
                <span
                  class="css-3ay1af"
                />
              </div>
            </dl>
          </div>
        </div>
        <br />
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h2
              class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
            >
              Transport details
            </h2>
            <dl
              class="css-39qxwj"
              id="transport-details"
            >
              <div
                class="css-16k4qo5"
                id="destination-country"
              >
                <dt
                  class="css-a1pm7p"
                >
                  What is the export destination?
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  Brazil
                </dd>
                <span
                  class="css-3ay1af"
                >
                  <a
                    class="css-80kuzi-StyledHoc e109g54z0"
                    href="/create-storage-document/GBR-2020-SD-123456/what-export-destination"
                    id="change-exportedTo"
                  >
                    Change
                    <span
                      class="govuk-visually-hidden"
                    >
                      destination country
                    </span>
                  </a>
                </span>
              </div>
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  How will the export leave the UK?
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  Truck
                </dd>
                <span
                  class="css-3ay1af"
                >
                  <a
                    class="css-80kuzi-StyledHoc e109g54z0"
                    href="/create-storage-document/GBR-2020-SD-123456/how-does-the-export-leave-the-uk"
                    id="change-transportType"
                  >
                    Change
                    <span
                      class="govuk-visually-hidden"
                    >
                      transport type
                    </span>
                  </a>
                </span>
              </div>
              <div
                class="css-16k4qo5"
              >
                <dt
                  class="css-a1pm7p"
                >
                  Do you have a road transport document to go with this export?
                </dt>
                <dd
                  class="css-1eokjdt"
                >
                  Yes
                </dd>
                <span
                  class="css-3ay1af"
                >
                  <a
                    class="css-80kuzi-StyledHoc e109g54z0"
                    href="/create-storage-document/GBR-2020-SD-123456/do-you-have-a-road-transport-document"
                    id="change-cmrType"
                  >
                    Change
                    <span
                      class="govuk-visually-hidden"
                    >
                      road transport document inclusion
                    </span>
                  </a>
                </span>
              </div>
            </dl>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="warning-message css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <div
              class="e1vcssft0 css-jxvtpr-StyledWarningText-StyledHoc e4u1u2g0"
            >
              <div
                class="css-96kz36-IconImportantWrapper e4u1u2g1"
              >
                <svg
                  fill="currentColor"
                  height="100%"
                  preserveAspectRatio="xMidYMid meet"
                  style="display: block;"
                  version="1.1"
                  viewBox="0 0 35.000000 35.000000"
                  width="100%"
                >
                  <title>
                    icon important
                  </title>
                  <g
                    transform="translate(0.000000,35.000000) scale(0.100000,-0.100000)"
                  >
                    <path
                      d="M100 332 c-87 -48 -125 -155 -82 -232 48 -87 155 -125 232 -82 87 48
125 155 82 232 -48 87 -155 125 -232 82z m100 -122 c0 -53 -2 -60 -20 -60 -18
0 -20 7 -20 60 0 53 2 60 20 60 18 0 20 -7 20 -60z m0 -111 c0 -12 -7 -19 -20
-19 -19 0 -28 28 -14 43 11 11 34 -5 34 -24z"
                    />
                  </g>
                </svg>
              </div>
              <strong
                class="css-jegmsb-WarningTextWrapper e4u1u2g2"
              >
                In submitting the information in this storage document, I acknowledge that I understand that I am under a duty not to knowingly submit information which I know to be false or otherwise inaccurate. The fisheries administrations may undertake checks on the accuracy and validity of the data and may cross reference it with data held and submitted by others in the supply chain, and may exchange data with other relevant fisheries administrations, including those in the country of export. I understand that I may be liable to investigation and potentially to enforcement action, including prosecution if I knowingly submit false, misleading or otherwise inaccurate data.
              </strong>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <button
              class="button button-start"
              id="continue"
              type="submit"
            >
              Accept and create storage document
            </button>
          </div>
        </div>
      </form>
      <br />
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
